kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -274,6 +275,34 @@ public static KernelException invalidConfigurationValueException(
     return new InvalidConfigurationValueException(key, value, helpMessage);
   }
 
+  public static KernelException domainMetadataUnsupported() {
+    String message =
+        "Found DomainMetadata action(s) but table feature 'domainMetadata' "
+            + "is not supported on this table.";
+    return new KernelException(message);
+  }
+
+  public static KernelException duplicateDomainMetadataAction(
+      String domain, String action1, String action2) {
```

Comment:
Do you want to print the entire DomainMetadata? (like, just DomainMetadata::toString?) Just the domain instead? Please use a more specific variable than `action1`.